crnk-core/src/main/java/io/crnk/core/engine/internal/jackson/JacksonResourceFieldInformationProvider.java

```diff
@@ -97,9 +97,9 @@ else if (JsonInclude.Include.NON_EMPTY.equals(value)) {
 
 	@Override
 	public Optional<String> getJsonName(BeanAttributeInformation attributeDesc) {
```

Comment:
not sure, but maybe this method gets obsoleted entirely by the (Property)NamingResolver

Reply:
Not sure about this though.. issue is that `JacksonResourceFieldInformationProvider` holds not only json-property name resolver but also jackson-naming-strategy related thing.. moving that path to `BeanAttributeInformation` would require it to have an instance of `ObjectMapper` or `BeanInformationFactory.ObjectMapper`. Are you ok with this?

Reply:
how do you mean ` jackson-naming-strategy related thing.. `?

Currently `ResourceInformationProviderBase` looks like: 

```
protected String getJsonName(BeanAttributeInformation attributeDesc, ResourceFieldType fieldType) {
        if (fieldType == ResourceFieldType.ID && enforceIdName) {
            // referred to as id even if named differently in Java
            return "id";
        }
        for (ResourceFieldInformationProvider fieldInformationProvider : resourceFieldInformationProviders) {
            Optional<String> jsonName = fieldInformationProvider.getJsonName(attributeDesc);
            if (jsonName.isPresent()) {
                return jsonName.get();
            }
        }
        return attributeDesc.getName();
    }
```

Here it could rather make use of `attributeDesc.getJsonName`. Then the `JacksonResourceFieldInformationProvider.getJsonName` should not be necessary I think. Was hoping for the `BeanFactory` to be independent of `ObjectMapper`. But maybe a missed something.

Reply:
It doesn't work this way unfortunately.. `JacksonResourceFieldInformationProvider` holds information about the current jackson naming strategy and therefor removing `JacksonResourceFieldInformationProvider.getJsonName` from snippet you posted would actually cause that crnk will loose its ability to respect jackson naming strategy property... meaning each `ResourceInformationProviderBase. getJsonName` would return either property default name (in case no JacksonProperty annotation used) or the value from that annotation (calculated by new JacksonNameResolver). Which means naming strategy will be totally ignored

Reply:
maybe let's focus on the comment above https://github.com/crnk-project/crnk-framework/pull/743#discussion_r434329831, currently we may not speak about the same things :-)